.pre-commit-config.yaml

```diff
@@ -39,3 +39,11 @@ repos:
         language: script
         pass_filenames: false
         description: "Runs `golangci-lint`, requires https://github.com/golangci/golangci-lint"
+  - repo: https://github.com/python-jsonschema/check-jsonschema
```

Comment:
This hook will validate zarf.yaml against the given schema, which is not what we are looking for in #216. The hook should automatically modify zarf.schema.json if the codebase has changed the schema. The way hooks like that work is they create the file and diff it against the one in the codebase. If they aren't identical, it applies the changes to the codebase and fails the hook. The `terraform-docs` hook in [this repo](https://github.com/antonbabenko/pre-commit-terraform#terraform_docs) is a good example of what we want. Here's [the code](https://github.com/antonbabenko/pre-commit-terraform/blob/master/hooks/terraform_docs.sh) for that hook.

Reply:
I do see this hook as being valuable though, just not as a fix to #216

Reply:
I updated the the story, I think it's fine to fail on pre-commit and warn them for now.  If we feel it needs to do more later on, we can always expand on it, but the intent is met to make sure we don't push up invalid type changes without also the schema updates.

Reply:
> As a developer of Zarf I'd like a pre-commit hook to automatically verify the schema file is correct

I have no problem with that, though what this does is different from that. If the story was "As a developer of Zarf I'd like a pre-commit hook to automatically verify `zarf.yaml` against the existing schema file" this would be a good resolution to that.

Reply:
The codebase is the only source of truth for the schema definition. If the codebase changes, let's say by removing a previously existing field, but the schema isn't changed, `zarf.yaml` could still have that field that now doesn't exist and the check would pass even though the schema isn't correct any more.

This hook is good, but only as a protection against making mistakes in `zarf.yaml`.

Reply:
Oh I see what you're saying.  Yes that's a problem.  We need to make sure that the schema matches the types and I didn't realize this was just checking the `zarf.yaml`, which won't necessarily have every field anyway.  I think we'll need to change this to PR for that to work @YrrepNoj.

Reply:
Let's merge this PR, but leave #216 open. The combination of #216 and this will be strong, by first verifying that the schema is up to date, then verifying zarf.yaml against the now-known-good schema